### PR TITLE
remove connected disconnected state, 3rd attempt

### PR DESCRIPTION
**Pull request by drahnr**
Split commits into tiny bit-sized chunks.

Re-introduces functionality of #3868 without the previous issue.

### CODE REVIEW

**Comment by drahnr** [comment]
Commit `9ac71f5` verified to work.

File: node/service/src/lib.rs [hunk 1]
```diff
@@ -850,8 +903,10 @@ where
 		local_keystore.and_then(move |k| authority_discovery_service.map(|a| (a, k)));
 
 	let overseer_handle = if let Some((authority_discovery_service, keystore)) = maybe_params {
+		// already have access to the handle
```
Comment (ordian) [inline node/service/src/lib.rs:906]: outdated comment?
```suggestion
```

**Review by ordian — APPROVED** [review]
Looks good, but I suggest we confirm it works on a validator as well as a full node before merging. And CI needs fixing.

**Comment by drahnr** [comment]
Checked `dfe7044`

**Review by eskimor — APPROVED** [review]

**Review by eskimor — APPROVED** [review]
Just to recap - why is it not possible to not construct the overseer in the first place, if we find that we are not a relay chain? And make the handle so that it cannot exist without an overseer - so chain selection cannot be wrong?

**Comment by drahnr** [comment]
> Just to recap - why is it not possible to not construct the overseer in the first place, if we find that we are not a relay chain? And make the handle so that it cannot exist without an overseer - so chain selection cannot be wrong?

It's not that simple. There is a rather large dependency loop, with chain selection, grandpa, substrate, and the overseer handle closing the loop. Now to resolve this the `OverseerConnector` was  created. An alternative way of fixing it alltogether, is indeed splitting the code into two distinct paths, but that itself is a rather invasive thing to do and we have to maintain two code paths for init or cut more granular. Out of scope of this PR, created https://github.com/paritytech/polkadot/issues/3951 to track this in for the future™.

**Comment by eskimor** [comment]
@ordian if there were no issues in testing, I would say - let's merge.

**Comment by ordian** [comment]
> Just to recap - why is it not possible to not construct the overseer in the first place, if we find that we are not a relay chain? And make the handle so that it cannot exist without an overseer - so chain selection cannot be wrong?

We were worried that disabling overseer on polkadot would interfere with block production when implementing #3321. So went with enabling overseer, but disabling approval-voting finality rule instead.

**Pull request merged**